File: src/test/scala/com/exasol/parquetio/TestFileManager.scala
package com.exasol.parquetio

import java.io.IOException
import java.nio.file._
import java.nio.file.attribute.BasicFileAttributes

/**
 * A mixin trait with file creation and deletion operations.
 */
trait TestFileManager {

  /**
   * Create a temporary folder.
   *
   * @param name the name of the directory.
   */
  final def createTemporaryFolder(name: String): Path =
    Files.createTempDirectory(name)

  /**
   * Recursively deletes files from the path.
   *
   * @param dir the path of a file or a directory.
   */
  final def deletePathFiles(dir: Path): Unit = {
    Files.walkFileTree(
      dir,
      new SimpleFileVisitor[Path] {
        override def visitFile(file: Path, attrs: BasicFileAttributes): FileVisitResult = {
          Files.delete(file)
          FileVisitResult.CONTINUE
        }
        override def postVisitDirectory(dir: Path, exc: IOException): FileVisitResult = {
          Files.delete(dir)
          FileVisitResult.CONTINUE
        }
      }
    )
    ()
  }

}

File: src/test/java/com/exasol/parquetio/TestFileManager.java
package com.exasol.parquetio;

import java.io.IOException;
import java.nio.file.*;
import java.nio.file.attribute.BasicFileAttributes;

final class TestFileManager {
    private TestFileManager() {
        // utility class
    }

    static Path createTemporaryFolder(final String name) throws IOException {
        return Files.createTempDirectory(name);
    }

    static void deletePathFiles(final Path directory) throws IOException {
        Files.walkFileTree(directory, new SimpleFileVisitor<>() {
            @Override
            public FileVisitResult visitFile(final Path file, final BasicFileAttributes attributes)
                    throws IOException {
                Files.delete(file);
                return FileVisitResult.CONTINUE;
            }

            @Override
            public FileVisitResult postVisitDirectory(final Path directory, final IOException exception)
                    throws IOException {
                Files.delete(directory);
                return FileVisitResult.CONTINUE;
            }
        });
    }
}
